include/powsybl/iidm/Ref.hpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_REF_HPP
+#define POWSYBL_IIDM_REF_HPP
+
+namespace powsybl {
+
+namespace iidm {
+
+class Network;
+
+/**
+ * A {@link Ref} instance holds a reference to an object.
+ * It uses to maintain the back reference between an object and the parent it belongs to.
+ */
+template <typename T>
+class Ref {
```

Comment:
Make this class a real template (like in Java) because it is needed to store a `VariantManagerHolder` reference from `VariantArray`